src/data/polynomial/laurent.lean

```diff
@@ -294,6 +294,22 @@ left_inverse_trunc_to_laurent.injective
   f.to_laurent = g.to_laurent ↔ f = g :=
 ⟨λ h, polynomial.to_laurent_injective h, congr_arg _⟩
 
+instance : module R[X] R[T;T⁻¹] :=
+module.comp_hom _ polynomial.to_laurent
+
 end semiring
 
+section comm_semiring
+
+instance algebra_polynomial (R : Type*) [comm_semiring R] : algebra R[X] R[T;T⁻¹] :=
+{ commutes' := λ f l, by simp [mul_comm],
+  smul_def' := λ f l, rfl,
+  .. ((map_domain_ring_hom R int.of_nat_hom).comp
+    (to_finsupp_iso R).to_ring_hom : R[X] →+* R[T;T⁻¹]) }
+
+instance (R : Type*) [comm_semiring R] : is_scalar_tower R[X] R[X] R[T;T⁻¹] :=
```

Comment:
This shouldn't need commutativity I don't think

Reply:
You are right about this: I pushed a new version without the commutativity assumption.